<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ArrayWindow</class>
 <widget class="QMainWindow" name="ArrayWindow">
  <property name="windowModality">
   <enum>Qt::WindowModal</enum>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>996</width>
    <height>585</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>500</width>
    <height>500</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>100000</width>
    <height>100000</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Массив</string>
  </property>
  <property name="styleSheet">
   <string notr="true">	background-color: #171b22;
	color: #fff;</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout_3">
    <item>
     <layout class="QVBoxLayout" name="verticalLayout_3" stretch="0,0">
      <property name="spacing">
       <number>6</number>
      </property>
      <property name="leftMargin">
       <number>6</number>
      </property>
      <property name="topMargin">
       <number>6</number>
      </property>
      <property name="rightMargin">
       <number>6</number>
      </property>
      <property name="bottomMargin">
       <number>6</number>
      </property>
      <item>
       <widget class="QGroupBox" name="groupBox">
        <property name="styleSheet">
         <string notr="true">background-color: #0d1117; border: 0px; border-radius: 10px;</string>
        </property>
        <property name="title">
         <string>Массив</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <item>
          <widget class="QPlainTextEdit" name="dataArray">
           <property name="styleSheet">
            <string notr="true">background-color: transparent; color: #fff;</string>
           </property>
           <property name="placeholderText">
            <string>Массив данных</string>
           </property>
          </widget>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_2">
           <property name="spacing">
            <number>0</number>
           </property>
           <property name="leftMargin">
            <number>0</number>
           </property>
           <property name="topMargin">
            <number>0</number>
           </property>
           <property name="rightMargin">
            <number>0</number>
           </property>
           <property name="bottomMargin">
            <number>0</number>
           </property>
           <item>
            <widget class="QPushButton" name="randomDataButton">
             <property name="minimumSize">
              <size>
               <width>200</width>
               <height>30</height>
              </size>
             </property>
             <property name="maximumSize">
              <size>
               <width>200</width>
               <height>30</height>
              </size>
             </property>
             <property name="styleSheet">
              <string notr="true">	background-color: #7289DA;
	selection-background-color: #7289DA;
	color: #fff;
	border-width: 2px;
	border-style: solid;
	border-radius: 10px;
	alignment: center;
	border-color: #0d1117;</string>
             </property>
             <property name="text">
              <string>Заполнить случайными числами</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="readDataButton">
             <property name="minimumSize">
              <size>
               <width>200</width>
               <height>30</height>
              </size>
             </property>
             <property name="maximumSize">
              <size>
               <width>200</width>
               <height>30</height>
              </size>
             </property>
             <property name="styleSheet">
              <string notr="true">	background-color: #7289DA;
	selection-background-color: #7289DA;
	color: #fff;
	border-width: 2px;
	border-style: solid;
	border-radius: 10px;
	alignment: center;
	border-color: #0d1117;</string>
             </property>
             <property name="text">
              <string>Прочитать массив из файла</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout">
           <property name="leftMargin">
            <number>0</number>
           </property>
           <property name="topMargin">
            <number>0</number>
           </property>
           <property name="rightMargin">
            <number>0</number>
           </property>
           <property name="bottomMargin">
            <number>0</number>
           </property>
           <item>
            <widget class="QLabel" name="label_3">
             <property name="minimumSize">
              <size>
               <width>0</width>
               <height>30</height>
              </size>
             </property>
             <property name="maximumSize">
              <size>
               <width>16777215</width>
               <height>30</height>
              </size>
             </property>
             <property name="text">
              <string>Количество элементов: </string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="maxItem">
             <property name="minimumSize">
              <size>
               <width>100</width>
               <height>30</height>
              </size>
             </property>
             <property name="maximumSize">
              <size>
               <width>100</width>
               <height>30</height>
              </size>
             </property>
             <property name="styleSheet">
              <string notr="true">QSpinBox {
	background-color: #7289DA;
	selection-background-color: #7289DA;
	color: #fff;
	border-width: 2px;
	border-style: solid;
	border-radius: 10px;
	alignment: center;
	border-color: #0d1117;
}

QSpinBox::up-button, QSpinBox::down-button
{
    width: 12px;
    height: 12px;
    border: 0px solid #ccc;
    border-radius: 4px;
    background-color: #b0b0b0;
}</string>
             </property>
             <property name="maximum">
              <number>1000000000</number>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_4">
           <property name="leftMargin">
            <number>0</number>
           </property>
           <property name="topMargin">
            <number>0</number>
           </property>
           <property name="rightMargin">
            <number>0</number>
           </property>
           <property name="bottomMargin">
            <number>0</number>
           </property>
           <item>
            <widget class="QLabel" name="label_4">
             <property name="minimumSize">
              <size>
               <width>0</width>
               <height>30</height>
              </size>
             </property>
             <property name="maximumSize">
              <size>
               <width>16777215</width>
               <height>30</height>
              </size>
             </property>
             <property name="text">
              <string>Заполнить значениями от</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="minValue">
             <property name="minimumSize">
              <size>
               <width>100</width>
               <height>30</height>
              </size>
             </property>
             <property name="maximumSize">
              <size>
               <width>100</width>
               <height>30</height>
              </size>
             </property>
             <property name="styleSheet">
              <string notr="true">QSpinBox {
	background-color: #7289DA;
	selection-background-color: #7289DA;
	color: #fff;
	border-width: 2px;
	border-style: solid;
	border-radius: 10px;
	alignment: center;
	border-color: #0d1117;
}

QSpinBox::up-button, QSpinBox::down-button
{
    width: 12px;
    height: 12px;
    border: 0px solid #ccc;
    border-radius: 4px;
    background-color: #b0b0b0;
}</string>
             </property>
             <property name="minimum">
              <number>-1000000000</number>
             </property>
             <property name="maximum">
              <number>1000000000</number>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_5">
             <property name="minimumSize">
              <size>
               <width>30</width>
               <height>30</height>
              </size>
             </property>
             <property name="maximumSize">
              <size>
               <width>30</width>
               <height>30</height>
              </size>
             </property>
             <property name="text">
              <string>до</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignCenter</set>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="maxValue">
             <property name="minimumSize">
              <size>
               <width>100</width>
               <height>30</height>
              </size>
             </property>
             <property name="maximumSize">
              <size>
               <width>100</width>
               <height>30</height>
              </size>
             </property>
             <property name="styleSheet">
              <string notr="true">QSpinBox {
	background-color: #7289DA;
	selection-background-color: #7289DA;
	color: #fff;
	border-width: 2px;
	border-style: solid;
	border-radius: 10px;
	alignment: center;
	border-color: #0d1117;
}

QSpinBox::up-button, QSpinBox::down-button
{
    width: 12px;
    height: 12px;
    border: 0px solid #ccc;
    border-radius: 4px;
    background-color: #b0b0b0;
}</string>
             </property>
             <property name="minimum">
              <number>-1000000000</number>
             </property>
             <property name="maximum">
              <number>1000000000</number>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="groupBox_2">
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>0</height>
         </size>
        </property>
        <property name="styleSheet">
         <string notr="true">background-color: #0d1117; border: 0px; border-radius: 10px;</string>
        </property>
        <property name="title">
         <string>Обработка массива</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_4">
         <item>
          <widget class="QLabel" name="label">
           <property name="minimumSize">
            <size>
             <width>0</width>
             <height>30</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>16777215</width>
             <height>30</height>
            </size>
           </property>
           <property name="styleSheet">
            <string notr="true">background-color: transparent; color: #fff;</string>
           </property>
           <property name="text">
            <string>Количество элементов, кратных 5:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="label_2">
           <property name="minimumSize">
            <size>
             <width>0</width>
             <height>30</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>16777215</width>
             <height>30</height>
            </size>
           </property>
           <property name="text">
            <string>Сумма второго и наибольшего положительного числа массива:</string>
           </property>
          </widget>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_5">
           <item>
            <widget class="QPushButton" name="startProcess">
             <property name="minimumSize">
              <size>
               <width>150</width>
               <height>30</height>
              </size>
             </property>
             <property name="maximumSize">
              <size>
               <width>150</width>
               <height>30</height>
              </size>
             </property>
             <property name="styleSheet">
              <string notr="true">	background-color: #7289DA;
	selection-background-color: #7289DA;
	color: #fff;
	border-width: 2px;
	border-style: solid;
	border-radius: 10px;
	alignment: center;
	border-color: #0d1117;</string>
             </property>
             <property name="text">
              <string>Обработать</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QVBoxLayout" name="verticalLayout" stretch="0,0">
      <property name="spacing">
       <number>12</number>
      </property>
      <property name="leftMargin">
       <number>6</number>
      </property>
      <property name="topMargin">
       <number>6</number>
      </property>
      <property name="rightMargin">
       <number>6</number>
      </property>
      <property name="bottomMargin">
       <number>6</number>
      </property>
      <item>
       <widget class="QGroupBox" name="groupBox_3">
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>250</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>16777215</width>
          <height>250</height>
         </size>
        </property>
        <property name="styleSheet">
         <string notr="true">background-color: #0d1117; border: 0px; border-radius: 10px;</string>
        </property>
        <property name="title">
         <string>Сортировка и поиск</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_5">
         <item>
          <layout class="QVBoxLayout" name="verticalLayout_8">
           <property name="spacing">
            <number>12</number>
           </property>
           <item>
            <widget class="QRadioButton" name="isNextSearch">
             <property name="text">
              <string>Посделовательный поиск</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QRadioButton" name="isBinarySearch">
             <property name="text">
              <string>Бинарный поиск</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="dataForSearch">
             <property name="minimumSize">
              <size>
               <width>0</width>
               <height>30</height>
              </size>
             </property>
             <property name="maximumSize">
              <size>
               <width>16777215</width>
               <height>30</height>
              </size>
             </property>
             <property name="styleSheet">
              <string notr="true">	background-color: #7289DA;
	selection-background-color: #7289DA;
	color: #fff;
	border-width: 2px;
	border-style: solid;
	border-radius: 10px;
	alignment: center;
	border-color: #0d1117;</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignCenter</set>
             </property>
             <property name="placeholderText">
              <string>Данные для поиска</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="arraySearchingButton">
             <property name="minimumSize">
              <size>
               <width>150</width>
               <height>30</height>
              </size>
             </property>
             <property name="maximumSize">
              <size>
               <width>10000</width>
               <height>30</height>
              </size>
             </property>
             <property name="styleSheet">
              <string notr="true">	background-color: #7289DA;
	selection-background-color: #7289DA;
	color: #fff;
	border-width: 2px;
	border-style: solid;
	border-radius: 10px;
	alignment: center;
	border-color: #0d1117;</string>
             </property>
             <property name="text">
              <string>Найти</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QVBoxLayout" name="verticalLayout_6">
           <item>
            <widget class="QRadioButton" name="isInsertionSort">
             <property name="styleSheet">
              <string notr="true">
QCheckBox {
	color: #fff;
}
QCheckBox::indicator {
	border-radius: 7px;
	width: 15px;
	height: 15px;
}
QCheckBox::indicator:unchecked {
	border-radius: 7px;
	background-color: #171b22;
}
QCheckBox::indicator:unchecked:hover {
	border-radius: 7px;
	background-color: #171b22;
}

QCheckBox::indicator:unchecked:pressed {
	border-radius: 7px;
	background-color: #171b22;
}

QCheckBox::indicator:checked {
	border-radius: 7px;
	background-color: #c3e88d;
}

QCheckBox::indicator:checked:hover {
	border-radius: 7px;
	background-color: #c3e88d;
}

QCheckBox::indicator:checked:pressed {
	border-radius: 7px;
	background-color: #c3e88d;
}

QCheckBox::indicator:indeterminate:hover {
	border-radius: 7px;
	background-color: #c3e88d;
}

QCheckBox::indicator:indeterminate:pressed {
	border-radius: 7px;
	background-color: #c3e88d;
}</string>
             </property>
             <property name="text">
              <string>Сортировка вставками</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QRadioButton" name="isQuickSort">
             <property name="styleSheet">
              <string notr="true">
QCheckBox {
	color: #fff;
}
QCheckBox::indicator {
	border-radius: 7px;
	width: 15px;
	height: 15px;
}
QCheckBox::indicator:unchecked {
	border-radius: 7px;
	background-color: #171b22;
}
QCheckBox::indicator:unchecked:hover {
	border-radius: 7px;
	background-color: #171b22;
}

QCheckBox::indicator:unchecked:pressed {
	border-radius: 7px;
	background-color: #171b22;
}

QCheckBox::indicator:checked {
	border-radius: 7px;
	background-color: #c3e88d;
}

QCheckBox::indicator:checked:hover {
	border-radius: 7px;
	background-color: #c3e88d;
}

QCheckBox::indicator:checked:pressed {
	border-radius: 7px;
	background-color: #c3e88d;
}

QCheckBox::indicator:indeterminate:hover {
	border-radius: 7px;
	background-color: #c3e88d;
}

QCheckBox::indicator:indeterminate:pressed {
	border-radius: 7px;
	background-color: #c3e88d;
}</string>
             </property>
             <property name="text">
              <string>Быстрая сортировка</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <widget class="QPushButton" name="startSortingButton">
           <property name="minimumSize">
            <size>
             <width>150</width>
             <height>30</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>16777215</width>
             <height>30</height>
            </size>
           </property>
           <property name="styleSheet">
            <string notr="true">	background-color: #7289DA;
	selection-background-color: #7289DA;
	color: #fff;
	border-width: 2px;
	border-style: solid;
	border-radius: 10px;
	alignment: center;
	border-color: #0d1117;</string>
           </property>
           <property name="text">
            <string>Отсортировать</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="groupBox_4">
        <property name="styleSheet">
         <string notr="true">background-color: #0d1117; border: 0px; border-radius: 10px;</string>
        </property>
        <property name="title">
         <string>Отсортированный массив</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_7">
         <item>
          <widget class="QPlainTextEdit" name="sortedArray">
           <property name="toolTip">
            <string>Тут будет отсортированный массив
</string>
           </property>
           <property name="styleSheet">
            <string notr="true">background-color: transparent; color: #fff;</string>
           </property>
           <property name="placeholderText">
            <string>Отсортированный массив</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
